File: 110-balanced-binary-tree/balanced-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int height(TreeNode* node){
        if(node == NULL){
            return 0;
        }
        else{
            return 1 + max(height(node->left),height(node->right));
        }
    }
    bool isBalanced(TreeNode* root) {
        if(root==NULL){
            return true;
        }

        if(abs(height(root->left)-height(root->right))>1){
            return false;
        }
        if(!isBalanced(root->left) || !isBalanced(root->right)){
            return false;
        }
        return true;
    }
};

File: 110-balanced-binary-tree/balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int height(TreeNode node){
        if(node==null){
            return 0;
        }
        
        return 1+Math.max(height(node.left),height(node.right));
    }
    public boolean isBalanced(TreeNode root) {
        if(root==null){
            return true;
        }
        int l = height(root.left);
        int r = height(root.right);
        if(l>r+1 || r>l+1){
            return false;
        }
        return isBalanced(root.left) && isBalanced(root.right);
    }
}